src/utils/daos_hdlr.c

```diff
@@ -2528,92 +2532,202 @@ fs_copy(struct file_dfs *src_file_dfs,
 	return rc;
 }
 
+inline void
+set_dm_args_default(struct dm_args *dm)
+{
+	dm->src = NULL;
+	dm->dst = NULL;
+	uuid_clear(dm->src_p_uuid);
+	uuid_clear(dm->src_c_uuid);
+	uuid_clear(dm->dst_p_uuid);
+	uuid_clear(dm->dst_c_uuid);
+	dm->src_poh = DAOS_HDL_INVAL;
+	dm->src_coh = DAOS_HDL_INVAL;
+	dm->dst_poh = DAOS_HDL_INVAL;
+	dm->dst_coh = DAOS_HDL_INVAL;
+}
+
 static int
-fs_copy_connect(struct file_dfs *src_file_dfs,
-		struct file_dfs *dst_file_dfs,
-		struct fs_copy_args *fa,
-		char *sysname,
-		daos_cont_info_t *src_cont_info,
-		daos_cont_info_t *dst_cont_info,
-		struct duns_attr_t *src_dattr,
-		struct duns_attr_t *dst_dattr)
+dm_get_cont_type(struct dm_args *ca,
+		 char *sysname,
+		 daos_cont_info_t *cont_info,
+		 daos_cont_layout_t *type)
 {
-	/* check source pool/conts */
-	int rc = 0;
+	int                     rc = 0;
+	daos_prop_t*            prop = daos_prop_alloc(1);
+
+	if (prop == NULL) {
+		fprintf(stderr, "Failed to allocate prop (%d)", rc);
+		D_GOTO(out, rc);
+	}
 
+	prop->dpp_entries[0].dpe_type = DAOS_PROP_CO_LAYOUT_TYPE;
+
+	rc = daos_cont_query(ca->src_coh, NULL, prop, NULL);
+	if (rc) {
+		fprintf(stderr, "daos_cont_query() failed (%d)", rc);
+		daos_prop_free(prop);
+		D_GOTO(out, rc);
+	}
+
+	*type = prop->dpp_entries[0].dpe_val;
+	daos_prop_free(prop);
+out:
+	return rc;
+}
+
+static int
+dm_connect(bool is_posix_copy,
+     	   struct file_dfs *src_file_dfs,
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs